plugins/core-plugin/src/main/resources/Dockerfile-template-yaml

```diff
@@ -1,16 +1,68 @@
-FROM ${baseContainerImage}
+#===================================================================#
+# Create build environment from base Python template launcher image #
+#===================================================================#
+FROM gcr.io/dataflow-templates-base/python311-template-launcher-base:latest as python-base
 
-# Copy template files to /template
+# Build args
 ARG WORKDIR=/template
-COPY main.py /template
-COPY requirements.txt /template
-COPY ${yamlTemplateName} /template/template.yaml
+ARG REQUIREMENTS_FILE=requirements.txt
+ARG BEAM_VERSION=${beamVersion}
+ARG BEAM_PACKAGE=apache-beam[dataframe,gcp,test,yaml]==$BEAM_VERSION
+ARG PY_VERSION=${pythonVersion}
+
+# Copy template files to /template
+RUN mkdir -p $WORKDIR
+COPY main.py requirements.txt* /template/
+<#if copyOtherFiles??>COPY ${copyOtherFiles} /template/</#if>
 WORKDIR $WORKDIR
 
-ENV FLEX_TEMPLATE_PYTHON_REQUIREMENTS_FILE=requirements.txt
+# Create requirements.txt file if not provided
+RUN if ! [ -f requirements.txt ] ; then echo "$BEAM_PACKAGE" > requirements.txt ; fi
+
+# Install dependencies to launch the pipeline and download to reduce startup time
+RUN python -m venv /venv \
+    && /venv/bin/pip install --no-cache-dir --upgrade pip setuptools \
+    && /venv/bin/pip install --no-cache-dir -U -r $REQUIREMENTS_FILE \
+    && /venv/bin/pip download --no-cache-dir --dest /tmp/dataflow-requirements-cache -r $REQUIREMENTS_FILE \
+    && rm -rf /usr/local/lib/python$PY_VERSION/site-packages  \
+    && mv /venv/lib/python$PY_VERSION/site-packages /usr/local/lib/python$PY_VERSION/
+
+
+#============================================================#
+# Create Distroless xlang image compatible with YamlTemplate #
+#============================================================#
+FROM ${baseJavaContainerImage}
```

Comment:
Looks like you changed python but not java, but similar (if less strong) assumptions about its structure seem to be built in. I'd vote we change this, but I won't block on it